### PR TITLE
updates workflow and adds custom formatting for rustfmt

### PR DESCRIPTION
**Pull request by ozgunozerk**
@nazar-pc , this `exclude` operation on matrices may be useful in monorepo as well maybe?

### CODE REVIEW

**Review by i1i1 — APPROVED** [review]
LGTM!

**Comment by ozgunozerk** [comment]
doesn't work though :D
yml file has an error again 🤦

**Comment by ozgunozerk** [comment]
Yay! It works

File: .github/workflows/ci-tests.yml [hunk 1]
```diff
@@ -25,6 +32,13 @@ jobs:
           - ubuntu-22.04
           - macos-12
           - windows-2022
+        run-all: 
+          - ${{ inputs.test-macos-and-windows == true || github.ref == 'refs/heads/main' }}
```
Comment (nazar-pc) [inline .github/workflows/ci-tests.yml:36]: If this actually works, it is indeed cleaner and I'd like to see it in monorepo.
I was trying to conditionally skip the job, but `input` wasn't available there.

Comment (nazar-pc) [reply .github/workflows/ci-tests.yml:36]: Yes, please :pray:

**Review by i1i1 — APPROVED** [review]

**Review by achiurizo — APPROVED** [review]
🧹

**Pull request merged**